astropy/units/format/cds.py

```diff
@@ -153,10 +153,10 @@ def p_main(p):
                  | factor
             """
             from astropy.units import dex
-            from astropy.units.core import Unit
+            from astropy.units.core import CompositeUnit, Unit
 
             if len(p) == 3:
-                p[0] = Unit(p[1] * p[2])
+                p[0] = CompositeUnit(p[1], p[2].bases, p[2].powers)
```

Comment:
I think this should be `p[1] * p[2].scale` (possibly, the scale can never be anything but 1, but might as well be sure not to discard non-unity scale, and it should not matter for the speed-up).